core/crypto/src/signature.rs

```diff
@@ -314,6 +314,16 @@ impl SecretKey {
     }
 }
 
+impl std::fmt::Display for SecretKey {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> Result<(), std::fmt::Error> {
```

Comment:
why do you need to display SecretKey? I specifically avoided implementation so it's harder to leak it.

Reply:
The `genesis-tools/keypair-generator` allows users to generate a bunch of key pairs (e.g. for pasting them into the csv file) currently it displays them into the terminal (later it will dump them into a password-protected file).

The way to implement a trait that allows a struct to be converted into a string is to use `ToString` trait (https://doc.rust-lang.org/std/string/trait.ToString.html) but it is not intended to be implemented directly, but instead one should implement `Display` and `ToString` will be automatically implemented.

Also, if we are concerned with `SecretKey` being displayed in some logs, not implementing `Display` does not help much because it can be displayed with `Debug`, it is only `{}` vs `{:?}` difference.